src/script/interpreter.cpp

```diff
@@ -1106,8 +1110,64 @@ class CTransactionSignatureSerializer {
 
 } // anon namespace
 
-uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType)
+uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, int sigversion)
 {
+    if (sigversion == 1) {
+        uint256 hashPrevouts;
+        uint256 hashSequence;
+        uint256 hashOutputs;
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY)) {
+            CHashWriter ss(SER_GETHASH, 0);
+            for (unsigned int n = 0; n < txTo.vin.size(); n++) {
+                ss << txTo.vin[n].prevout;
+            }
+            hashPrevouts = ss.GetHash(); // TODO: cache this value for all signatures in a transaction
+        }
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY) && (nHashType & 0x1f) != SIGHASH_SINGLE && (nHashType & 0x1f) != SIGHASH_NONE) {
```

Comment:
Also for BIP68: without signing nSequence of other inputs, other people may: 1. reduce the locktime to make an invalid signature valid; or 2. increase the locktime to make a valid signature invalid.

In the first case, you could just set your own nSequence to make sure the tx won't be mined until your preset time. So you have the full control.

In the second case, since the original signature is still valid, it could be confirmed immediately. So nothing changed